### ISSUE TITLE
Year conversion for kanji years

### ISSUE THREAD

**Issue by nicolasmaia** [birtles/rikaichamp#373]
It would be nice if Rikaichamp could also convert years displayed as 一九三七年 for example. Since the 年 is always present, I don't think we need to worry about false positives.

**Comment by birtles** [comment]
Sounds like a good idea! I wonder if there are other situations where converting numbers that are not years is helpful too?

**Comment by SaltfishAmi** [comment]
> 
> 
> Sounds like a good idea! I wonder if there are other situations where converting numbers that are not years is helpful too?

maybe just convert all kanji numbers longer than, say, three or four digits?

Oh and I wonder that, is it intended behavior to convert 令和七九年 to 令和63年 (7x9=63)??

**Comment by birtles** [comment]
> Oh and I found a bug that 令和七九年 will be converted to 令和63年 (7x9=63??), and 令和七二一三五六四九年 to 令和43年.

Yes, the number conversion code doesn't handle numbers that are transliterations of the digits. It only deals with 令和七十九年 etc.

**Comment by birtles** [comment]
> Yes, the number conversion code doesn't handle numbers that are transliterations of the digits. It only deals with 令和七十九年 etc.

Specifically, see comment here: https://github.com/birtles/rikaichamp/blob/5583183fe2b052ec4e49291d9205f60b899575dd/src/numbers.ts#L5-L6
